src/Simplex/Chat/View.hs

```diff
@@ -802,49 +802,47 @@ viewContactUpdated
     where
       fullNameUpdate = if T.null fullName' || fullName' == n' then " removed full name" else " updated full name: " <> plain fullName'
 
-viewReceivedMessage :: StyledString -> [StyledString] -> MsgContent -> CIMeta d -> [StyledString]
-viewReceivedMessage from quote mc meta = receivedWithTime_ from quote meta (ttyMsgContent mc)
+viewReceivedMessage :: StyledString -> [StyledString] -> MsgContent -> UTCTime -> CIMeta d -> [StyledString]
+viewReceivedMessage from quote mc ts meta = receivedWithTime_ ts from quote meta (ttyMsgContent mc)
 
-receivedWithTime_ :: StyledString -> [StyledString] -> CIMeta d -> [StyledString] -> [StyledString]
-receivedWithTime_ from quote CIMeta {localItemTs, createdAt} styledMsg = do
-  prependFirst (formattedTime <> " " <> from) (quote <> prependFirst indent styledMsg)
+receivedWithTime_ :: UTCTime -> StyledString -> [StyledString] -> CIMeta d -> [StyledString] -> [StyledString]
+receivedWithTime_ ts from quote CIMeta {localItemTs} styledMsg = do
+  prependFirst (ttyMsgTime ts localItemTs <> " " <> from) (quote <> prependFirst indent styledMsg)
   where
     indent = if null quote then "" else "      "
-    formattedTime :: StyledString
-    formattedTime =
-      let localTime = zonedTimeToLocalTime localItemTs
-          tz = zonedTimeZone localItemTs
-          format =
-            if (localDay localTime < localDay (zonedTimeToLocalTime $ utcToZonedTime tz createdAt))
-              && (timeOfDayToTime (localTimeOfDay localTime) > (6 * 60 * 60 :: DiffTime))
-              then "%m-%d" -- if message is from yesterday or before and 6 hours has passed since midnight
-              else "%H:%M"
-       in styleTime $ formatTime defaultTimeLocale format localTime
-
-viewSentMessage :: StyledString -> [StyledString] -> MsgContent -> CIMeta d -> [StyledString]
-viewSentMessage to quote mc = sentWithTime_ (prependFirst to $ quote <> prependFirst indent (ttyMsgContent mc))
+
+ttyMsgTime :: UTCTime -> ZonedTime -> StyledString
+ttyMsgTime ts t =
```

Comment:
it became harder to understand this without reading function's code and checking what's passed at call sites - change variable names to something more clear? Maybe this:
```suggestion
ttyMsgTime currentTs localItemTs =
```

Reply:
I think it's ok, it's consistently used in all functions...